### PR TITLE
Some fixes to unbreak main

### PR DESCRIPTION
**Pull request by tigarmo**
- [ ] Have you followed the guidelines for contributing?
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `tox`?

-----

### CODE REVIEW

File: craft_application/remote/git.py [hunk 1]
```diff
@@ -221,7 +221,7 @@ def _init_repo(self) -> None:
                 f"Could not initialize a git repository in {str(self.path)!r}."
             ) from error
 
-    def push_url(  # pylint: disable=too-many-branches
+    def push_url(  # noqa: PLR0912 (too many branches)
```
Comment (dariuszd21) [inline craft_application/remote/git.py:224]:
```suggestion
    def push_url(  # noqa: PLR0912 (too-many-branches)
```
I saw that usually we use dashed version of the rule

File: pyproject.toml [hunk 2]
```diff
@@ -54,6 +54,8 @@ dev = [
     "pytest-mock==3.14.0",
     "pytest-rerunfailures==14.0",
     "pytest-time>=0.3.1",
+    # Pin requests because of https://github.com/msabramo/requests-unixsocket/issues/73
```
Comment (mr-cal) [inline pyproject.toml:57]: Which upstream bug is more correct, that one or this one? https://github.com/psf/requests/issues/6707

**Review by mr-cal — APPROVED** [review]

File: pyproject.toml (hunk 2 above)
Comment (tigarmo) [reply pyproject.toml:57]: as far as I can tell this is not a requests bug but an apparently breaking change that will stay as is:
image
so if my read of all this is correct, this needs to be fixed in requests-unixsocket

Comment (mr-cal) [reply pyproject.toml:57]: Makes sense!

**Review by lengau — APPROVED** [review]

**Pull request merged**